zio-kafka/src/main/scala/zio/kafka/consumer/internal/Runloop.scala

```diff
@@ -153,9 +154,22 @@ private[consumer] final class Runloop private (
     }
   }
 
-  private val getConsumerGroupMetadataIfAny: UIO[Option[ConsumerGroupMetadata]] =
-    if (settings.hasGroupId) consumer.runloopAccess(c => ZIO.attempt(c.groupMetadata())).fold(_ => None, Some(_))
-    else ZIO.none
+  /**
+   * @return
+   *   optionally consumer group metadata, first we try get it from consumerMetadataRef, or fetch it from consumer if
+   *   not present. If the group id is not set, we return None.
+   */
+  private def getConsumerGroupMetadataIfAny: UIO[Option[ConsumerGroupMetadata]] =
```

Comment:
There is no need to make this a `def`, it can stay a `val`. ZIO workflows are values!

```suggestion
  private val getConsumerGroupMetadataIfAny: UIO[Option[ConsumerGroupMetadata]] =
```

Reply:
okay, it sounds a bit confusing. Because, as a beginner, I would guess vals would not be computed again each time. But I made changes to be consistent everywhere

Reply:
That is correct. The ZIO value, a workflow, is only constructed once, and then executed as many times as needed.